[abseil] Fix activation of `cxx17` feature

PigeonF opened the pull request:
Previously nothing would happen when specifying the `cxx17` feature, because the detection in the portfile is wrong.

PigeonF committed:
[abseil] Fix activation of `cxx17` feature

Previously nothing would happen when specifying the `cxx17` feature, because the detection in the portfile is wrong.

dg0yt commented:
> I am unsure whether or not I have to specify the `port-version` in the `vcpkg.json` file of the abseil port.

Yes. Basically, this is the file which you edit manually. The files in the `versions` directory is to be updated with `vcpkg x-add-version [--overwrite-version] abseil`. (The extra parameter is needed when you already added the new port version.

dg0yt commented on ports/abseil/portfile.cmake:
```diff
@@ -12,11 +12,16 @@ vcpkg_from_github(
 # uses the compiler default behavior to update `absl/base/options.h` as needed.
-if (ABSL_USE_CXX17 IN_LIST FEATURES)
+if (ABSL_USE_CXX17)
     set(ABSL_USE_CXX17_OPTION "-DCMAKE_CXX_STANDARD=17")
```
:+1: 
CC @coryan

PigeonF commented:
I have added the `port-version` in the `vcpk.json` and force pushed.

The `vcpkg x-add-version --all abseil` seems to have also changed `baseline.json`, is this expected?

PigeonF commented:
@microsoft-github-policy-service agree

PigeonF commented:
I don't understand what the CI check is complaining about. The output only says that the job failed with exit code 1. Is there some way to get more information about the failure?

anders-wind commented:
Our build is broken because of this, so we are looking forward to getting this in

coryan commented:
> I don't understand what the CI check is complaining about. The output only says that the job failed with exit code 1. Is there some way to get more information about the failure?

You can download the log files from:

https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/86191/artifacts?artifactName=failure%20logs%20for%20x64-osx&api-version=7.0&%24format=zip

I found that link at:

https://dev.azure.com/vcpkg/public/_build/results?buildId=86191&view=artifacts&pathAsName=false&type=publishedArtifacts

from:

https://dev.azure.com/vcpkg/public/_build/results?buildId=86191&view=logs&jobId=7b75bd19-17d3-53d4-00fd-23f1a49a8ba4

anders-wind commented:
So it looks like osx failed with something unrelated?
```
-- Found OpenCV: /Users/vagrant/Data/installed/x64-osx (found version "4.7.0") 
CMake Error at include/dv-processing/CMakeLists.txt:2 (CONFIGURE_FILE):
  No such file or directory


-- Boost::stacktrace will provide basic info.
-- Configuring incomplete, errors occurred!
```

PigeonF commented:
So, just to clarifiy, I should merge master and push again? Not rebase my commit onto master and force push?

dg0yt commented:
The result is equivalent.
